### PR TITLE
Actionable Improvements

### PR DESCRIPTION
**Pull request by nevillco**
In working on a client project, we have Services that require delegation and would benefit from Actionable. We soon realized:

- The Services framework does not include Actionable
- Even if it did, the existing implementation breaks for non-`internal` types

This PR:

- Tweaks the Actionable template to use the access control of the implementing type
- Moves "utility" functions into a new `Actionable+Helpers.swifttemplate` file for readability
- Adds a Sourcery build phase to the Services target, and moves the template files (which are now shared between targets) to Services
- Allows for nested Actionable types. Prior to this PR, the following code would cause Sourcery to generate non-compiling code:
```swift
class Foo {
    class Bar: Actionable {
        weak var delegate: Delegate?
        enum Action {
            case baz
        }
    }
}
```
Now, it generates:
```swift
// MARK: - Foo.Bar
internal protocol FooBarDelegate: class {

    func bar(_ component: Foo.Bar, didNotify action: Foo.Bar.Action)

}

internal extension Foo.Bar {

    typealias ActionType = Action
    typealias Delegate = FooBarDelegate

    func notify(_ action: ActionType) {
        delegate?.bar(self, didNotify: action)
    }

    func with(delegate: Delegate) -> Self {
        self.delegate = delegate
        return self
    }

}
```

### CODE REVIEW

**Comment by heyltsjay** [comment]
@nevillco Agreed that we shouldn't duplicate the template files. I would be ok with them living in their `Services` home if you can pull off the generation. Or perhaps an alias can help.

**Comment by nevillco** [comment]
@heyltsjay Done - template files are living in Services

File: PRODUCTNAME/app/PRODUCTNAME/Resources/Generated/Actionable+AutoConformance.swift [hunk 1]
```diff
@@ -19,14 +21,21 @@ extension AuthCoordinator {
         delegate?.authCoordinator(self, didNotify: action)
     }
 
+    func with(delegate: Delegate) -> Self {
```
Comment (heyltsjay) [inline PRODUCTNAME/app/PRODUCTNAME/Resources/Generated/Actionable+AutoConformance.swift]: @nevillco maybe a touch opinionated for the template?

Comment (nevillco) [reply PRODUCTNAME/app/PRODUCTNAME/Resources/Generated/Actionable+AutoConformance.swift]: Fair :+1:

**Comment by nevillco** [comment]
Closing in favor of #116

**Pull request closed without merging**